src/guru.jl

```diff
@@ -70,13 +70,16 @@ julia> p = finufft_makeplan(3,2,+1,1,1e-4,dtype=Float32,nthreads=4);
 creates a plan for a 2D type 3 Float32 transform with tolerance 1e-4,
 to use 4 threads.
 """
-function finufft_makeplan(type::Integer,
+function finufft_makeplan(type::Integer, args...; dtype=Float64, kwargs...)
+    finufft_makeplan(dtype, type, args...; kwargs...)
+end
```

Comment:
Is there any measurable benefit from this change? If I understand it correctly, it allows the compiler to move the `if dtype==Float64` from the function body to the multiple dispatch system?

If this is useful, I think that the helper function should be internal and have a unique name so that it doesn't clutter the exported namespace. Also, collecting the arguments into `args...` hides useful information from the data shown by `methods(finufft_makeplan)`